src/models/user.js

```diff
@@ -0,0 +1,56 @@
+const allUsers = []
+class User {
+  constructor(email = null, password = null , userName = null) {
+    this.email = email;
+    this.password = password;
+    this.userName = userName;
+
+  }
+
+  addUser(){
+    let id = allUsers.length ? allUsers[allUsers.length - 1].id + 1 : 1
+    allUsers.push({
+      id,
+      email: this.email,
+      password: this.password,
+      userName: this.userName
+    })
+    return {
+      id,
+      email: this.email,
+      password: this.password,
+      userName: this.userName
+
+    }
+  }
+  getAllUsers(){
+    return allUsers
+  }
+
+  getSingleUser(reqData){
+    const {key, value}= reqData;
+    const user = allUsers.find(user=>{user[key]=== value})
+    return user
+  }
+  updateSingleUser(reqData){
+    const {key, value} = reqData
+  const singleUser = allUsers.find(user => user.key === value )
+  return singleUser
+  }
+
+
+  deleteUser(id){
+   const user= allUsers.filter(id)
```

Comment:
For delete you should just have allUsers.filter(user => user.id !== id), But do a validation in your middleware to check if it exists.